plugins/node/instrumentation-undici/src/types.ts

```diff
@@ -51,18 +52,29 @@ export interface RequestHookFunction<T = UndiciRequest> {
   (span: Span, request: T): void;
 }
 
+export interface ResponseHookFunction<
+  RequestType = UndiciResponse,
+  ResponseType = UndiciResponse
+> {
+  (span: Span, request: RequestType, response: ResponseType): void;
```

Comment:
Please use an `Info` object to store all relevant hook data. 
In the future, more arguments might be added which will make the usage harder.

See example [here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/80304bcd1bb4654499fb4f67e804d142f0284d21/plugins/node/instrumentation-amqplib/src/types.ts#L44)

Reply:
If I may ask something is to keep the generics since they give flexibility to the devs to pass the type specific to the `undici` version they're using